evervault/errors/error_handler.py

```diff
@@ -1,7 +1,40 @@
 from . import evervault_errors as errors
 
 
-def raise_errors_on_failure(resp, body=None):
+def raise_errors_on_function_run_failure(function_body):
+    error = function_body.get("error")
+    if error:
+        message = error.get("message")
+        stack = error.get("stack")
+        id = function_body.get("id")
+        raise errors.FunctionRuntimeError(message, stack, id)
+    raise errors.UnexpectedError(
+        "An unexpected error occurred. Please contact Evervault support"
+    )
+
+
+def raise_errors_on_api_error(resp, response_body):
+    if resp.status_code >= 200 and resp.status_code < 300:
+        return
+    code = response_body.get("code")
+    detail = response_body.get("detail")
+
+    if code == "unauthorized":
```

Comment:
Not sure there is any benefit on introducing that many errors for non retriable errors. I'd just keep `FunctionNotReadyError` & `FunctionTimeoutError` and default on `EvervaultError` for the rest

Reply:
These aren't errors I introduced, they exist already in the SDK. I agree that they're not especially helpful for users, but it seemed odd for the error to exist but not used universally. Is the move here to remove its use everywhere in the SDK?

Reply:
Per offline, these changes will be made in a subsequent PR